src/components/fb/FidelityBondSteps.tsx

```diff
@@ -0,0 +1,290 @@
+        on the length.
+      </div>
+      <LockdateForm onChange={(date) => onDateSelected(date)} yearsRange={selectableYearsRange} />
+    </div>
+  )
+}
+
+const SelectJar = ({ accountBalances, totalBalance, utxos, selectedJar, onJarSelected }: SelectJarProps) => {
+  const sortedAccountBalances: Array<AccountBalanceSummary> = useMemo(() => {
+    if (!accountBalances) return []
+    return Object.values(accountBalances).sort((lhs, rhs) => lhs.accountIndex - rhs.accountIndex)
+  }, [accountBalances])
+
+  return (
+    <div className="d-flex flex-column gap-4">
+      <div className={styles.stepDescription}>Select a jar to fund the fidelity bond from.</div>
+      <div className="d-flex justify-content-between">
+        {sortedAccountBalances.map((account, index) => (
+          <SelectableJar
+            key={index}
+            index={account.accountIndex}
+            balance={account.totalBalance}
+            isSelectable={utxos[account.accountIndex] && utxos[account.accountIndex].length > 0}
+            isSelected={selectedJar === account.accountIndex}
+            fillLevel={calculateFillLevel(Number.parseFloat(account.totalBalance), totalBalance)}
+            onClick={() => onJarSelected(account.accountIndex)}
+          />
+        ))}
+      </div>
+    </div>
+  )
+}
+
+const UtxoCard = ({
+  utxo,
+  isSelectable = true,
+  isSelected = false,
+  isLoading = false,
+  onClick = () => {},
+}: UtxoCardProps) => {
+  const settings = useSettings()
+
+  return (
+    <div
+      className={cx('utxoCard', { selected: isSelected, selectable: isSelectable })}
+      onClick={() => isSelectable && onClick()}
+    >
+      <div className={styles.utxoSelectionMarker}>
+        {isSelected && <Sprite symbol="checkmark" width="18" height="18" />}
+      </div>
+      <div className={styles.utxoBody}>
+        <Balance valueString={utxo.value.toString()} convertToUnit={settings.unit} showBalance={true} />
+        <code className={styles.utxoAddress}>{utxo.address}</code>
+        <div className={styles.utxoDetails}>
+          <div>{utxo.path}</div>
+          <div>&#183;</div>
+          <div>{utxo.confirmations} Confirmations</div>
+        </div>
+      </div>
+      {isLoading && (
+        <div className={styles.utxoLoadingSpinner}>
+          <rb.Spinner as="span" animation="border" size="sm" role="status" aria-hidden="true" />
+        </div>
+      )}
+      {!isLoading && utxo.frozen && !utxo.locktime && (
+        <div className={cx('utxoLabel', 'utxoFrozen')}>
+          <Sprite symbol="lock" width="18" height="18" />
+          <div>frozen</div>
+        </div>
+      )}
+      {!isLoading && utxo.locktime && (
+        <div className={cx('utxoLabel', 'utxoFidelityBond')}>
+          <Sprite symbol="lock" width="18" height="18" />
+          <div>locked</div>
+        </div>
+      )}
+      {!isLoading && !utxo.frozen && utxo.label === 'cj-out' && (
```

Comment:
Just noticed that the classification (`new`, `deposit`, `cj-out`, etc.) are not provided in the data from `/utxo`, but only in `/display`, where they are not on a "utxo", but on address level..

Nonetheless, I think displaying the label here (even when it cannot be set by Jam yet), should be displayed here.

This is an object in `entries` of the `display` response:
```json
{
	"0": {
		"hd_path": "m/84'/1'/1'/1/0",
		"address": "bcrt1qeme6gzjlj5s95jfe8hwauagy8y45teh4rej8jw",
		"status": "cj-out",
		"label": "",
		"extradata": "",
		"amount": "149.99994620",
		"available_balance": "149.99994620"
	}
}
```
Compare with the utxo data structure (no `status` here):
```json
{
	"0": {
		"address": "bcrt1qeme6gzjlj5s95jfe8hwauagy8y45teh4rej8jw",
		"path": "m/84'/1'/1'/1/0",
		"label": "",
		"value": 14999994620,
		"tries": 1,
		"tries_remaining": 2,
		"external": false,
		"mixdepth": 1,
		"confirmations": 7,
		"frozen": false,
		"utxo": "9dbd6db083b3fb23c936bd16dbc52efa64330a3145d4c8c9492136a00c8e4c80:1"
	}
}
```

Reply:
If I understand correctly, the API does not return the UTXO labels in the UTXO endpoint but only for the `/display` wallet endpoint on an address label and we need to manually match up UTXOs and addresses to find out which UTXO has which label? 🤔

Reply:
> If I understand correctly, the API does not return the UTXO labels in the UTXO endpoint but only for the `/display` wallet endpoint on an address label and we need to manually match up UTXOs and addresses to find out which UTXO has which label? thinking


Correct.


> Do you know what the difference between a label and a status is?

A `label` is manually applied by the user (https://github.com/JoinMarket-Org/joinmarket-clientserver/blob/3140616d322c85e71591d8d4d46cbb4cea374002/docs/release-notes/release-notes-0.9.4.md#user-chosen-address-labelling). Whereas the `status` a metric applied by the backend (https://github.com/theborakompanioni/joinmarket-clientserver/blob/36bf36bd65f6365cb282a8abf3512e8e11df548a/jmclient/jmclient/wallet_utils.py#L493-L531).